extensions/amp-intersection-observer-polyfill/0.1/amp-intersection-observer-polyfill.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {installIntersectionObserver} from 'intersection-observer/intersection-observer.install';
+import {registerServiceBuilder} from '../../../src/service';
+import {upgradePolyfill} from '../../../src/polyfillstub/intersection-observer-stub';
+
+const TAG = 'amp-intersection-observer-polyfill';
+
+/**
+ * @param {!Window} win
+ * @return {!Object}
+ */
+function install(win) {
+  upgradePolyfill(win, () => {
+    installIntersectionObserver();
+    // TODO(dvoytenko): hookup 3p host updater for inabox.
+  });
+  return {};
+}
+
+/**
+ * Registers the polyfill.
+ * @param {!Window} win
+ */
+export function installIntersectionObserverPolyfill(win) {
```

Comment:
Another "install". :) Does this need export? Can it just be inlined below?

Reply:
I think the issue is that `install()` (now renamed to `upgrade()`) function must be a strictly factory `function` that we can call a `new` on. If I inline it, someone can replace the `function()` with `() => {}` and break the upgrade path. We debugged something like this with @erwinmombay before and it was pretty painful.